### PR TITLE
ServiceNow V2 - Added a better file handling

### PR DESCRIPTION
**Pull request by reutshal**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/33167

## Description
Added a file handling with the getFilePath method in case the file was not found in the context.

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 1]
```diff
@@ -1173,9 +1173,10 @@ def upload_file_command(client: Client, args: dict) -> Tuple[str, Dict, Dict, bo
...
-    file_name = args.get('file_name', demisto.dt(demisto.context(), "File(val.EntryID=='" + file_id + "').Name"))
-    if not file_name:  # in case of info file
-        file_name = demisto.dt(demisto.context(), "InfoFile(val.EntryID=='" + file_id + "').Name")
+    file_name = args.get('file_name')
+    if not file_name:
+        file_data = demisto.getFilePath(file_id)
+        file_name = file_data.get('name')
     if not file_name:
         raise Exception('Could not find the file. Please add a file to the incident.')
     file_name = file_name[0] if isinstance(file_name, list) else file_name
```
Comment (yaakovi) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:1182]: this line can be removed now

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 2]
```diff
@@ -1173,9 +1173,10 @@ def upload_file_command(client: Client, args: dict) -> Tuple[str, Dict, Dict, bo
     ticket_id = str(args.get('id', ''))
     file_id = str(args.get('file_id', ''))
 
-    file_name = args.get('file_name', demisto.dt(demisto.context(), "File(val.EntryID=='" + file_id + "').Name"))
-    if not file_name:  # in case of info file
-        file_name = demisto.dt(demisto.context(), "InfoFile(val.EntryID=='" + file_id + "').Name")
+    file_name = args.get('file_name')
```
Comment (yaakovi) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:1176]: wasn't the argument used before?

Comment (reutshal) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:1176]: What do you mean used before? only if the arg was not present we searched the file name in the context

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py (hunk 1 above)
Comment (yaakovi) [reply Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:1181]: ?

**Review by yaakovi — APPROVED** [review]

**Pull request merged**